esd-group-3/src/java/models/Bookings.java

```diff
@@ -6,19 +6,90 @@
 package models;
 
 import dbcon.DBConnection;
+import java.sql.Date;
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.sql.Statement;
+import java.sql.Time;
+import java.text.SimpleDateFormat;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 import java.util.logging.Level;
 import java.util.logging.Logger;
-
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Calendar;
+import java.sql.Time;
+import java.time.LocalTime;
 /**
  *
  * @author conranpearce
  */
 public class Bookings {
     
+    // Create a booking by appending data to the bookingslot table
+    public void createBooking(DBConnection dbcon, String username, String employeeUsername, Date date, Time startTime, Time endTime) {
+        String query = "";
+
+        // Get the user's id depending on the username
+        query = "SELECT id FROM Users WHERE username = '" + username + "'";
+        int userid = 0;
+        try (Statement stmt = dbcon.conn.createStatement()) {
+            ResultSet resultSet = stmt.executeQuery(query);
+            while (resultSet.next()) {
+                userid = resultSet.getInt("id");
+            }
+        } catch (SQLException e) {
+            System.out.println(e);
+        }
+        
+        // Get the client id from the userid in the Clients table
+        query = "SELECT id FROM Clients WHERE userid = " + userid;
+        int clientid = 0;
+        try (Statement stmt = dbcon.conn.createStatement()) {
+            ResultSet resultSet = stmt.executeQuery(query);
+            while (resultSet.next()) {
+                clientid = resultSet.getInt("id");
+            }
+        } catch (SQLException e) {
+            System.out.println(e);
+        }
+        
+        // Get the id of the employee from the employee username
+        query = "SELECT id FROM Users WHERE username = '" + employeeUsername + "'";
+        int employeeidbefore = 0;
+        try (Statement stmt = dbcon.conn.createStatement()) {
+            ResultSet resultSet = stmt.executeQuery(query);
+            while (resultSet.next()) {
+                employeeidbefore = resultSet.getInt("id");
+            }
+        } catch (SQLException e) {
+            System.out.println(e);
+        }
```

Comment:
I think you can use the user Id from above instead of this block of code

Reply:
Resolved